src/react-native-animated-charts/src/charts/linear/ChartPath.tsx

```diff
@@ -106,297 +106,362 @@ function positionXWithMargin(x: number, margin: number, width: number) {
   }
 }
 
-export const ChartPath = React.memo(
-  ({
-    hapticsEnabled,
-    hitSlop = 0,
-    width,
-    height,
-    stroke = 'black',
-    selectedStrokeWidth = 1,
-    strokeWidth = 1,
-    gestureEnabled = true,
-    selectedOpacity = 0.7,
-    timingFeedbackConfig,
-    timingAnimationConfig,
-    longPressGestureHandlerProps = {},
-    ...props
-  }: ChartPathProps) => {
-    const {
-      positionX,
-      positionY,
-      originalX,
-      originalY,
-      state,
-      isActive,
-      progress,
-      pathOpacity,
-      currentPath,
-      previousPath,
-    } = useChartData();
-
-    const interpolatorWorklet = useWorkletValue();
-
-    const translationX = useSharedValue<number | null>(null);
-    const translationY = useSharedValue<number | null>(null);
-
-    const setOriginData = useWorkletCallback(
-      (path: PathData, index?: number) => {
-        if (!path.data.length) {
-          return;
-        }
-
-        if (typeof index === 'undefined') {
-          originalX.value = '';
-          originalY.value = '';
-          return;
-        }
-
-        originalX.value = path.data[index].x.toString();
-        originalY.value = path.data[index].y.toString();
-      },
-      []
-    );
+const ChartPathInner = ({
+  hitSlop = 0,
+  stroke = 'black',
+  selectedStrokeWidth = 1,
+  strokeWidth = 1,
+  gestureEnabled = true,
+  selectedOpacity = 0.7,
+  hapticsEnabled,
+
+  width,
+  height,
+
+  timingFeedbackConfig,
+  timingAnimationConfig,
+  longPressGestureHandlerProps,
+  positionX,
+  positionY,
+  originalX,
+  originalY,
+  state,
+  isActive,
+  progress,
+  pathOpacity,
+  currentPath,
+  previousPath,
+  ...props
+}: ChartPathProps & Omit<ChartData, 'data' | 'dotScale'>) => {
+  const interpolatorWorklet = useWorkletValue();
+
+  const translationX = useSharedValue<number | null>(null);
+  const translationY = useSharedValue<number | null>(null);
+
+  const setOriginData = useWorkletCallback((path: PathData, index?: number) => {
+    if (!path.data.length) {
+      return;
+    }
 
-    const resetGestureState = useWorkletCallback(() => {
+    if (typeof index === 'undefined') {
```

Comment:
```suggestion
    if (index === undefined) {
```
I think we don't this patter at all

Reply:
but i see this is only whitespace change and we already discussed this and you had some strong opinions about this so w/e